airflow/jobs/scheduler_job.py

```diff
@@ -403,6 +403,15 @@ def _executable_task_instances_to_queued(self, max_tis: int, session: Session =
                     # Many dags don't have a task_concurrency, so where we can avoid loading the full
                     # serialized DAG the better.
                     serialized_dag = self.dagbag.get_dag(dag_id, session=session)
+                    # If the dag is missing, fail the task and continue to the next task.
+                    if not serialized_dag:
+                        self.log.error(
+                            "DAG '%s' for task instance %s not found in serialized_dag table",
+                            dag_id,
+                            task_instance,
+                        )
+                        task_instance.set_state(State.FAILED, session=session)
```

Comment:
Is it possible to set the entire DAG to `FAILED` here? Depending on the trigger rules, there could be downstream tasks which still attempt to execute, which will also fail due to a missing DAG.

Reply:
> there could be downstream tasks which still attempt to execute, which will then be marked failed by the same check

I think this is a good thing in this case. This code is reached because the DAG declaring those tasks is gone, so it doesn’t make sense to execute those tasks IMO.

Reply:
I will opt for setting all scheduled tasks to None. I doubt that failing the DAG here will really fail it when there're tasks being executed in executor. There's a bug that when you mark a DAG as failed, it comes up again as running https://github.com/apache/airflow/issues/16078

So I propose to set all scheduled tasks to None. The scheduler will no longer move the task instances to scheduled when the dag can no longer be found and it makes sense to set it to None instead of failing it since the task instances won't have logs

Reply:
> I think this is a good thing in this case. This code is reached because the DAG declaring those tasks is gone, so it doesn’t make sense to execute those tasks IMO.

Yeah, I agree that they won't be executable, I'm just wondering if we can mark them all as failed in a single pass.

Reply:
Done